ext/opentelemetry-ext-system-metrics/src/opentelemetry/ext/system_metrics/__init__.py

```diff
@@ -0,0 +1,215 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Instrument to report system (CPU, memory, network) and
+process (CPU, memory, garbage collection) metrics. By default, the
+following metrics are configured:
+
+"system_memory": ["total", "available", "used", "free"],
+"system_cpu": ["user", "system", "idle"],
+"network_bytes": ["bytes_recv", "bytes_sent"],
+"runtime_memory": ["rss", "vms"],
+"runtime_cpu": ["user", "system"],
+
+
+Usage
+-----
+
+.. code:: python
+
+    from opentelemetry.ext.system_metrics import SystemMetrics
+    from opentelemetry.sdk.metrics.export import ConsoleMetricsExporter
+
+    exporter = ConsoleMetricsExporter()
+    SystemMetrics(exporter)
+
+    # metrics are collected asynchronously
+    input("...")
+
+API
+---
+"""
+
+import gc
+import os
+import typing
+
+import psutil
+
+from opentelemetry import metrics
+from opentelemetry.sdk.metrics.export import MetricsExporter
+from opentelemetry.sdk.metrics.export.controller import PushController
+
+
+class SystemMetrics:
+    def __init__(
+        self,
+        exporter: MetricsExporter,
+        interval: int = 30,
+        labels: typing.Optional[typing.Dict[str, str]] = None,
+        config: typing.Optional[typing.Dict[str, typing.List[str]]] = None,
+    ):
+        self._labels = {} if labels is None else labels
+        self.meter = metrics.get_meter(__name__)
+        self.controller = PushController(
+            meter=self.meter, exporter=exporter, interval=interval
+        )
+        if config is None:
```

Comment:
Is there somewhere that we can let the user know that passing in their own config is possible? Maybe an example file?